src/aggregate/aggregate_exec.c

```diff
@@ -33,6 +29,44 @@ typedef struct {
   const PLN_ArrangeStep *lastAstp;
 } cachedVars;
 
+static void reeval_key(RedisModuleCtx *outctx, const RSValue *key) {
+  RedisModuleString *rskey = NULL;
+  if (key) {
+    if(key->t == RSValue_Reference) {
```

Comment:
Technically, the `RSValue_Dereference` should be done in a loop no?

Reply:
it is a loop, no?
```
static inline RSValue *RSValue_Dereference(const RSValue *v) {
  for (; v && v->t == RSValue_Reference; v = v->ref)
    ;
  return (RSValue *)v;
}
```